pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -130,139 +158,207 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 		return reconcile.Result{}, err
 	}
 
-	// Fetch source provider secret
-	sourceProviderSecretObj, err := r.fetchSecret(instance)
+	// Init provider:
+	provider, err := r.initProvider(instance)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-
-	provider, err := r.createProvider(instance)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-
-	err = provider.Connect(sourceProviderSecretObj)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	defer provider.Close()
-
-	// Load source VM:
-	err = provider.LoadVM(instance.Spec.Source)
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-
-	// Load the external resource mapping
-	resourceMapping, err := r.fetchResourceMapping(instance.Spec.ResourceMapping, instance.Namespace)
+	mapper, err := provider.CreateMapper()
 	if err != nil {
-		//TODO: update Validating status condition
 		return reconcile.Result{}, err
 	}
 
-	// Prepare/merge the resourceMapping
-	provider.PrepareResourceMapping(resourceMapping, instance.Spec.Source)
-
-	// Validate if it's needed at this stage of processing
-	if shouldValidate(&instance.Status) {
-		conditions, err := provider.Validate()
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		err = r.upsertStatusConditions(request.NamespacedName, conditions)
+	vmName := types.NamespacedName{Name: instance.Status.TargetVMName, Namespace: request.Namespace}
+	if instance.Status.TargetVMName == "" {
+		// Define VM spec
+		template, err := provider.FindTemplate()
+		var spec = &kubevirtv1.VirtualMachine{}
```

Comment:
Spec is always assigned later; no need to initialize it here explicitly.

Reply:
Actually it is, because it's used outside if-else block.

Reply:
Yes, and it is assigned in every path of execution either from `ProcessTemplate` or `CreateEmptyVM`.

`var spec = &kubevirtv1.VirtualMachine{}` creates `VirtualMachine` and then assigns pointer to it to `spec`
vs. 
`var spec *kubevirtv1.VirtualMachine` is initialized with pointer's zero value - `nil`
And the `spec` pointer is always overwritten in if-else.

Reply:
Ah, I thought to declare it, sorry :) Fixed.